Build for visionOS using the public SDK

emw-apple opened the pull request:
#### 9c7eb634447a4afda39fb837b236fc545d430cc3
<pre>
Build for visionOS using the public SDK
<a href="https://bugs.webkit.org/show_bug.cgi?id=273517">https://bugs.webkit.org/show_bug.cgi?id=273517</a>
<a href="https://rdar.apple.com/problem/127315794">rdar://problem/127315794</a>

Reviewed by Tim Horton and Alexey Proskuryakov.

WebKit has built for visionOS for a long time, but relies on some
internal-only idioms that must be changed for public SDK compatibility:

- The public visionOS SDK does not fall back to iphoneos/iphonesimulator
  property conditions, so we need to add &quot;[sdk=xr*]&quot; clauses and handle
  a WK_PLATFORM_NAME of xros/xrsimulator in xcconfigs.

- TARGET_OS_IOS is not true, so a number of #if guards need to be
  updated to include TARGET_OS_VISION when it is available.

Beyond that, add needed SPI declarations, similar to any other embedded
platform update.

* Configurations/SDKVariant.xcconfig:
* Source/ThirdParty/ANGLE/Configurations/ANGLE-dynamic.xcconfig:
* Source/ThirdParty/libwebrtc/Configurations/Base-libwebrtc.xcconfig:
* Source/ThirdParty/libwebrtc/Configurations/DebugRelease.xcconfig:
* Source/ThirdParty/libwebrtc/Configurations/libaom.xcconfig:
* Source/ThirdParty/libwebrtc/Configurations/libvpx.xcconfig:
* Source/ThirdParty/libwebrtc/Configurations/libwebrtc.xcconfig:
* Source/ThirdParty/libwebrtc/Configurations/yasm.xcconfig:

* Source/WTF/wtf/PlatformEnableCocoa.h: SEPARATED_MODEL and
  MODEL_PROCESS features currently only build with WebKitAdditions, so
  temporarily disable them in the public build.
* Source/WTF/wtf/PlatformHave.h:
  - HAVE_LOCKDOWN_MODE_PDF_ADDITIONS: This was never enabled in internal
    builds, turn it off publicly.
  - HAVE_CORE_TELEPHONY: It&apos;s a nominally public framework but stripped
    from the public SDK. Rather than hack it back into place, disable
    the feature since it is minimally used for autofill.
  - HAVE_BROWSER_ENGINE_SUPPORTING_API: Only needed in public builds, so
    the PLATFORM(VISION) clause was missing until now.

* Source/WebCore/Configurations/FeatureDefines.xcconfig:
* Source/WebCore/Configurations/WebCore.xcconfig:
* Source/WebCore/Configurations/WebCoreTestSupport.xcconfig:
* Source/WebCore/PAL/Configurations/PAL.xcconfig:
* Source/WebCore/PAL/pal/spi/cocoa/AVFoundationSPI.h:
* Source/WebCore/PAL/pal/spi/cocoa/QuartzCoreSPI.h:
* Source/WebCore/PAL/pal/spi/ios/MobileGestaltSPI.h:
* Source/WebCore/PAL/pal/spi/ios/SystemPreviewSPI.h:
* Source/WebCore/platform/ios/WebItemProviderPasteboard.h:
* Source/WebKit/Configurations/FeatureDefines.xcconfig:
* Source/WebKit/Configurations/WebKit.xcconfig:
* Source/WebKit/Platform/spi/Cocoa/AuthenticationServicesCoreSPI.h:
* Source/WebKit/Platform/spi/visionos/RealitySystemSupportSPI.h: Added.
* Source/WebKit/Shared/API/Cocoa/_WKHitTestResult.h:
* Source/WebKit/UIProcess/API/Cocoa/WKUIDelegate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKUIDelegatePrivate.h: Drive-by
  removal of some outdated version checks.
* Source/WebKit/UIProcess/API/Cocoa/WKWebView.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewPrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebsiteDataStore.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKElementAction.h: Drive-by remove
  a defined(TARGET_OS_IOS) check, to make the preprocessor conditional
  easier to write. This file already requires TargetConditionals.h to be
  included, and TARGET_OS_IOS has existed for many years.
* Source/WebKit/UIProcess/API/Cocoa/_WKElementAction.mm: Ditto.
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeInteractionRegionLayers.mm:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKitLegacy/mac/WebView/WebUIDelegatePrivate.h:
* Source/WebKitLegacy/mac/WebView/WebViewPrivate.h:
* Tools/TestWebKitAPI/Configurations/TestWebKitAPI.xcconfig:
* Tools/TestWebKitAPI/Configurations/WebProcessPlugIn.xcconfig:

Check in TBD stubs for private frameworks, generated by
extract-tbds-from-internal-sdk.

* WebKitLibraries/SDKs/xros1.0-additions.sdk/SDKSettings.plist: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/SymlinkedHeaders-output.xcfilelist: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/SymlinkedHeaders.xcfilelist: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/AppServerSupport.framework/AppServerSupport.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/AppStoreDaemon.framework/AppStoreDaemon.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/AppSupport.framework/AppSupport.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/ApplePushService.framework/ApplePushService.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/AuthKit.framework/AuthKit.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/BackBoardServices.framework/BackBoardServices.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/CorePrediction.framework/CorePrediction.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/CoreRE.framework/CoreRE.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/FontServices.framework/libFontParser.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/FrontBoardServices.framework/FrontBoardServices.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/GraphicsServices.framework/GraphicsServices.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/IOSurfaceAccelerator.framework/IOSurfaceAccelerator.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/IconServices.framework/IconServices.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/IdleTimerServices.framework/IdleTimerServices.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/InstallCoordination.framework/InstallCoordination.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/LinearMediaKit.framework/LinearMediaKit.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/MRUIKit.framework/MRUIKit.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/MobileKeyBag.framework/MobileKeyBag.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/PrototypeTools.framework/PrototypeTools.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/RunningBoardServices.framework/RunningBoardServices.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/SafariSafeBrowsing.framework/SafariSafeBrowsing.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/URLFormatting.framework/URLFormatting.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/VisionKitCore.framework/VisionKitCore.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/usr/lib/swift/libswiftXPC.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/usr/lib/xr/libRuntimeSupport.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/usr/local/include/AvailabilityProhibitedInternal.h: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/SDKSettings.plist: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/SymlinkedHeaders-output.xcfilelist: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/SymlinkedHeaders.xcfilelist: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/System/Library/PrivateFrameworks/AppServerSupport.framework/AppServerSupport.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/System/Library/PrivateFrameworks/AppStoreDaemon.framework/AppStoreDaemon.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/System/Library/PrivateFrameworks/AppSupport.framework/AppSupport.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/System/Library/PrivateFrameworks/ApplePushService.framework/ApplePushService.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/System/Library/PrivateFrameworks/AuthKit.framework/AuthKit.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/System/Library/PrivateFrameworks/BackBoardServices.framework/BackBoardServices.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/System/Library/PrivateFrameworks/CorePrediction.framework/CorePrediction.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/System/Library/PrivateFrameworks/CoreRE.framework/CoreRE.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/System/Library/PrivateFrameworks/FontServices.framework/libFontParser.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/System/Library/PrivateFrameworks/FrontBoardServices.framework/FrontBoardServices.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/System/Library/PrivateFrameworks/GraphicsServices.framework/GraphicsServices.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/System/Library/PrivateFrameworks/IconServices.framework/IconServices.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/System/Library/PrivateFrameworks/IdleTimerServices.framework/IdleTimerServices.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/System/Library/PrivateFrameworks/InstallCoordination.framework/InstallCoordination.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/System/Library/PrivateFrameworks/LinearMediaKit.framework/LinearMediaKit.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/System/Library/PrivateFrameworks/MRUIKit.framework/MRUIKit.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/System/Library/PrivateFrameworks/PrototypeTools.framework/PrototypeTools.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/System/Library/PrivateFrameworks/RunningBoardServices.framework/RunningBoardServices.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/System/Library/PrivateFrameworks/SafariSafeBrowsing.framework/SafariSafeBrowsing.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/System/Library/PrivateFrameworks/URLFormatting.framework/URLFormatting.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/System/Library/PrivateFrameworks/VisionKitCore.framework/VisionKitCore.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/usr/lib/swift/libswiftGLKit.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/usr/lib/swift/libswiftXPC.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/usr/lib/xr/libRuntimeSupport.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/usr/local/include/AvailabilityProhibitedInternal.h: Added.

Canonical link: <a href="https://commits.webkit.org/278405@main">https://commits.webkit.org/278405@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/81dd3ca89f466b362ee694be39803231f0d6ea8b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/50263 "2 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/29556 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/2560 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/53522 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/953 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/52566 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/35784 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/592 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/41018 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/52362 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/27223 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/43264 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/22117 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/24639 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/517 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/8641 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/43592 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/46625 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/581 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/55107 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/49762 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/25359 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/514 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/48417 "Passed tests") | 
| | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/26622 "Build is in progress. Recent messages:OS: Sonoma (14.4.1), Xcode: 15.3; Skipping applying patch since patch_id isn't provided; Checked out pull request; Compiled WebKit (warnings)") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/43448 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/47443 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/11059 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/27484 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/57240 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/26352 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/11765 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on Source/WTF/wtf/PlatformEnableCocoa.h:
```diff
@@ -1014,7 +1014,7 @@
 #endif
 
-#if !defined(ENABLE_MODEL_PROCESS) && PLATFORM(VISION)
+#if !defined(ENABLE_MODEL_PROCESS) && PLATFORM(VISION) && USE(APPLE_INTERNAL_SDK)
```
This probably needs a bug to track enabling.

aproskuryakov commented on Source/WebCore/Configurations/WebCore.xcconfig:
```diff
@@ -196,6 +200,7 @@ SECTORDER_FLAGS_Production[sdk=macosx*] = -Wl,-order_file,WebCore.order;
 
 EXCLUDED_SOURCE_FILE_NAMES = pdfjs*/*;
 EXCLUDED_SOURCE_FILE_NAMES[sdk=iphone*] = $(inherited) *.tiff *Cursor.png npapi.h npfunctions.h npruntime.h npruntime_internal.h;
+EXCLUDED_SOURCE_FILE_NAMES[sdk=xr*] = $(inherited) *.tiff *Cursor.png npapi.h npfunctions.h npruntime.h npruntime_internal.h;
```
We still have the `np` files?!

emw-apple replied:
Doesn't look like we do. These names were just left behind from the headers were removed in 2021 in <https://commits.webkit.org/243690@main>.

emw-apple committed:
Build for visionOS using the public SDK
https://bugs.webkit.org/show_bug.cgi?id=273517
rdar://problem/127315794

Reviewed by Tim Horton and Alexey Proskuryakov.

WebKit has built for visionOS for a long time, but relies on some
internal-only idioms that must be changed for public SDK compatibility:

- The public visionOS SDK does not fall back to iphoneos/iphonesimulator
  property conditions, so we need to add "[sdk=xr*]" clauses and handle
  a WK_PLATFORM_NAME of xros/xrsimulator in xcconfigs.

- TARGET_OS_IOS is not true, so a number of #if guards need to be
  updated to include TARGET_OS_VISION when it is available.

Beyond that, add needed SPI declarations, similar to any other embedded
platform update.

* Configurations/SDKVariant.xcconfig:
* Source/ThirdParty/ANGLE/Configurations/ANGLE-dynamic.xcconfig:
* Source/ThirdParty/libwebrtc/Configurations/Base-libwebrtc.xcconfig:
* Source/ThirdParty/libwebrtc/Configurations/DebugRelease.xcconfig:
* Source/ThirdParty/libwebrtc/Configurations/libaom.xcconfig:
* Source/ThirdParty/libwebrtc/Configurations/libvpx.xcconfig:
* Source/ThirdParty/libwebrtc/Configurations/libwebrtc.xcconfig:
* Source/ThirdParty/libwebrtc/Configurations/yasm.xcconfig:

* Source/WTF/wtf/PlatformEnableCocoa.h: SEPARATED_MODEL and
  MODEL_PROCESS features currently only build with WebKitAdditions, so
  temporarily disable them in the public build.
* Source/WTF/wtf/PlatformHave.h:
  - HAVE_LOCKDOWN_MODE_PDF_ADDITIONS: This was never enabled in internal
    builds, turn it off publicly.
  - HAVE_CORE_TELEPHONY: It's a nominally public framework but stripped
    from the public SDK. Rather than hack it back into place, disable
    the feature since it is minimally used for autofill.
  - HAVE_BROWSER_ENGINE_SUPPORTING_API: Only needed in public builds, so
    the PLATFORM(VISION) clause was missing until now.

* Source/WebCore/Configurations/FeatureDefines.xcconfig:
* Source/WebCore/Configurations/WebCore.xcconfig:
* Source/WebCore/Configurations/WebCoreTestSupport.xcconfig:
* Source/WebCore/PAL/Configurations/PAL.xcconfig:
* Source/WebCore/PAL/pal/spi/cocoa/AVFoundationSPI.h:
* Source/WebCore/PAL/pal/spi/cocoa/QuartzCoreSPI.h:
* Source/WebCore/PAL/pal/spi/ios/MobileGestaltSPI.h:
* Source/WebCore/PAL/pal/spi/ios/SystemPreviewSPI.h:
* Source/WebCore/platform/ios/WebItemProviderPasteboard.h:
* Source/WebKit/Configurations/FeatureDefines.xcconfig:
* Source/WebKit/Configurations/WebKit.xcconfig:
* Source/WebKit/Platform/spi/Cocoa/AuthenticationServicesCoreSPI.h:
* Source/WebKit/Platform/spi/visionos/RealitySystemSupportSPI.h: Added.
* Source/WebKit/Shared/API/Cocoa/_WKHitTestResult.h:
* Source/WebKit/UIProcess/API/Cocoa/WKUIDelegate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKUIDelegatePrivate.h: Drive-by
  removal of some outdated version checks.
* Source/WebKit/UIProcess/API/Cocoa/WKWebView.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewPrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebsiteDataStore.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKElementAction.h: Drive-by remove
  a defined(TARGET_OS_IOS) check, to make the preprocessor conditional
  easier to write. This file already requires TargetConditionals.h to be
  included, and TARGET_OS_IOS has existed for many years.
* Source/WebKit/UIProcess/API/Cocoa/_WKElementAction.mm: Ditto.
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeInteractionRegionLayers.mm:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKitLegacy/mac/WebView/WebUIDelegatePrivate.h:
* Source/WebKitLegacy/mac/WebView/WebViewPrivate.h:
* Tools/TestWebKitAPI/Configurations/TestWebKitAPI.xcconfig:
* Tools/TestWebKitAPI/Configurations/WebProcessPlugIn.xcconfig:

Check in TBD stubs for private frameworks, generated by
extract-tbds-from-internal-sdk.

* WebKitLibraries/SDKs/xros1.0-additions.sdk/SDKSettings.plist: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/SymlinkedHeaders-output.xcfilelist: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/SymlinkedHeaders.xcfilelist: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/AppServerSupport.framework/AppServerSupport.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/AppStoreDaemon.framework/AppStoreDaemon.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/AppSupport.framework/AppSupport.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/ApplePushService.framework/ApplePushService.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/AuthKit.framework/AuthKit.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/BackBoardServices.framework/BackBoardServices.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/CorePrediction.framework/CorePrediction.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/CoreRE.framework/CoreRE.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/FontServices.framework/libFontParser.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/FrontBoardServices.framework/FrontBoardServices.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/GraphicsServices.framework/GraphicsServices.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/IOSurfaceAccelerator.framework/IOSurfaceAccelerator.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/IconServices.framework/IconServices.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/IdleTimerServices.framework/IdleTimerServices.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/InstallCoordination.framework/InstallCoordination.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/LinearMediaKit.framework/LinearMediaKit.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/MRUIKit.framework/MRUIKit.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/MobileKeyBag.framework/MobileKeyBag.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/PrototypeTools.framework/PrototypeTools.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/RunningBoardServices.framework/RunningBoardServices.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/SafariSafeBrowsing.framework/SafariSafeBrowsing.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/URLFormatting.framework/URLFormatting.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/System/Library/PrivateFrameworks/VisionKitCore.framework/VisionKitCore.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/usr/lib/swift/libswiftXPC.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/usr/lib/xr/libRuntimeSupport.tbd: Added.
* WebKitLibraries/SDKs/xros1.0-additions.sdk/usr/local/include/AvailabilityProhibitedInternal.h: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/SDKSettings.plist: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/SymlinkedHeaders-output.xcfilelist: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/SymlinkedHeaders.xcfilelist: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/System/Library/PrivateFrameworks/AppServerSupport.framework/AppServerSupport.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/System/Library/PrivateFrameworks/AppStoreDaemon.framework/AppStoreDaemon.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/System/Library/PrivateFrameworks/AppSupport.framework/AppSupport.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/System/Library/PrivateFrameworks/ApplePushService.framework/ApplePushService.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/System/Library/PrivateFrameworks/AuthKit.framework/AuthKit.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/System/Library/PrivateFrameworks/BackBoardServices.framework/BackBoardServices.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/System/Library/PrivateFrameworks/CorePrediction.framework/CorePrediction.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/System/Library/PrivateFrameworks/CoreRE.framework/CoreRE.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/System/Library/PrivateFrameworks/FontServices.framework/libFontParser.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/System/Library/PrivateFrameworks/FrontBoardServices.framework/FrontBoardServices.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/System/Library/PrivateFrameworks/GraphicsServices.framework/GraphicsServices.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/System/Library/PrivateFrameworks/IconServices.framework/IconServices.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/System/Library/PrivateFrameworks/IdleTimerServices.framework/IdleTimerServices.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/System/Library/PrivateFrameworks/InstallCoordination.framework/InstallCoordination.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/System/Library/PrivateFrameworks/LinearMediaKit.framework/LinearMediaKit.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/System/Library/PrivateFrameworks/MRUIKit.framework/MRUIKit.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/System/Library/PrivateFrameworks/PrototypeTools.framework/PrototypeTools.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/System/Library/PrivateFrameworks/RunningBoardServices.framework/RunningBoardServices.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/System/Library/PrivateFrameworks/SafariSafeBrowsing.framework/SafariSafeBrowsing.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/System/Library/PrivateFrameworks/URLFormatting.framework/URLFormatting.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/System/Library/PrivateFrameworks/VisionKitCore.framework/VisionKitCore.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/usr/lib/swift/libswiftGLKit.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/usr/lib/swift/libswiftXPC.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/usr/lib/xr/libRuntimeSupport.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator1.0-additions.sdk/usr/local/include/AvailabilityProhibitedInternal.h: Added.

Canonical link: https://commits.webkit.org/278405@main
